files/rsyslog/rsyslog.sh

```diff
@@ -45,15 +45,15 @@ if [ "${CDM_USE_UNDEFINED,,}" = "false" -a "${CDM_KEEP_EMPTY_FIELDS}" = "" -a "$
         if [ "${USE_MMEXTERNAL:-}" = "" ] ; then
             export SKIP_EMPTY=false
             export USE_MMEXTERNAL=true
-            echo "INFO: SKIP_EMPTY was set to false, but USE_MMEXTERNAL was not set and none of the UNDEFINED fields were configured. Enabling USE_MMEXTERNAL. Consider using SKIP_EMPTY=true instead."
+            echo "INFO: SKIP_EMPTY was set to false, but USE_MMEXTERNAL was not set and none of the environment variables to manage undefined and empty fields were configured. Enabling USE_MMEXTERNAL. Consider using SKIP_EMPTY=true instead."
         elif [ "${USE_MMEXTERNAL,,}" = "true" ] ; then
             export SKIP_EMPTY=false
             export USE_MMEXTERNAL=true
-            echo "INFO: SKIP_EMPTY was set to false and USE_MMEXTERNAL was set to true, but none of the UNDEFINED fields were configured. Consider using SKIP_EMPTY=true and unsetting USE_MMEXTERNAL instead."
+            echo "INFO: SKIP_EMPTY was set to false and USE_MMEXTERNAL was set to true, but none of the environment variables to manage undefined and empty fields were configured. Consider using SKIP_EMPTY=true and unsetting USE_MMEXTERNAL instead."
```

Comment:
_Consider using SKIP_EMPTY=true and unsetting USE_MMEXTERNAL instead_
Should you change this to suggest setting the CDM_* environment variables? If the user set `USE_MMEXTERNAL` wouldn't that suggest the user wants to use MMEXTERNAL, but neglected to set the variables?

_Consider setting the CDM_* environment variables or using SKIP_EMPTY=true and unsetting USE_MMEXTERNAL._

Reply:
Under these settings, no undefined field handling is not executed, but just empty fields are skipped.  When this is what a user wants to do, setting SKIP_EMPTY=true and use mmnormalize would be preferable for the performance reason.  (that's the reason why SKIP_EMPTY was introduced.)  So, if only empty fields are to be skipped, we'd like to recommend SKIP_EMPTY=true.